frappe/client.py

```diff
@@ -211,21 +204,12 @@ def insert_many(docs=None):
 	if isinstance(docs, str):
 		docs = json.loads(docs)
 
-	out = []
-
 	if len(docs) > 200:
 		frappe.throw(_("Only 200 inserts allowed in one request"))
 
+	out = []
```

Comment:
This won't work because each time the number of children changes, and the object is essentially different:

```python
In [1]: doc = frappe.get_doc("Sales Invoice", "SINV-21-00070")

In [2]: x = []

In [3]: x.append(doc)

In [4]: doc = frappe.get_doc("Sales Invoice", "SINV-21-00070")

In [5]: doc.append("items")
Out[5]: <SalesInvoiceItem: unsaved parent=SINV-21-00070>

In [6]: x
Out[6]: [<SalesInvoice: SINV-21-00070>]

In [7]: doc
Out[7]: <SalesInvoice: SINV-21-00070>

In [8]: doc in x
Out[8]: False

```

Might need a dict with `(dt, dn)` as key for this.

Reply:
hmm yep that would make sense if `insert_many` returned the whole list of doc objects, but since we finally only return the doc names ..set seems to be more suitable?

seems i missed out on this in my original review comment, apologies for that 😅

Reply:
right, sorry. should've read it better 👓

Reply:
yes, let's make it a set. Returning as set is also okay (optional), it will get auto-coerced in `frappe.as_json`